index.d.ts

```diff
@@ -0,0 +1,59 @@
+declare namespace DbTypes {
+    type ClientType = 'mysql' | 'sqlite' | 'postgres';
+
+    interface DbErrorOpts {
+        nativeError: {
+            message: string
+        }
+        client: ClientType
+    }
+
+    class DBError extends Error {
+        name: string;
+        nativeError: Error;
+        client: ClientType;
+
+        constructor(args?: DbErrorOpts)
```

Comment:
As mentioned above, this is probably useless. It would be pretty difficult to create a well defined error. Or do you have a use case in mind?

Reply:
Removed